### PR TITLE
Load project/package data from top-level directory

### PR DESCRIPTION
**Pull request by shelbyspees**
I expect this PR to include the following changes:
- [X] Un-nest Package from Project class
- [X] Factor out `json_inspect` function
- [x] Add `load_projects` function that reads in base path and returns list of Project objects

### CODE REVIEW

File: lib/revolution/project.rb [hunk 1]
```diff
@@ -5,30 +5,62 @@
...
+    projects.push(Project.new(path)) if File.exist?(path + '/recipe.rb')
+  end
+  projects
+end
+
 class Project
   attr_accessor :name, :data, :packages
 
   def initialize(path)
     @path = path
```
Comment (smaslennikov) [inline lib/revolution/project.rb]: Is this necessary? The function seems simple enough to use argument as local variables

File: spec/project_spec.rb [hunk 2]
```diff
@@ -4,8 +4,37 @@
...
+    end
+    it 'returns an array of one or more Project objects' do
+      expect(@projects[0]).to be_a Project
+    end
+  end
+end
+
 describe 'Project' do
-  context 'project with one recipe and no dependencies' do
+  context 'project with one package' do
```
Comment (smaslennikov) [inline spec/project_spec.rb]: Package is really our _end product_. Should this maybe be recipe instead?

File: lib/revolution/project.rb [hunk 3]
```diff
@@ -5,30 +5,62 @@
 require 'open3'
 require 'fpm/cookery/recipe'
 
+def json_inspect(path)
+  cmd = 'fpm-cook inspect ' + path
+  Open3.popen3(cmd) do |stdin, stdout, stderr, wait_thr|
+    pid = wait_thr.pid
+    stdin.close
+    json   = stdout.read
```
Comment (smaslennikov) [inline lib/revolution/project.rb]: I'm not sure this is a very descriptive variable name.

File: lib/revolution/project.rb [hunk 4]
```diff
@@ -5,30 +5,62 @@
...
+    err    = stderr.read
+    status = wait_thr.value
+    # puts err unless status.exitstatus.zero?
+    begin
+      data = JSON.parse(json)
+    rescue JSON::ParserError
+      # this should never happen
```
Comment (smaslennikov) [inline lib/revolution/project.rb]: But if it does, maybe yell loudly at the user?

File: lib/revolution/project.rb (hunk 3 above)
Comment (shelbyspees) [reply lib/revolution/project.rb]: This comment prompted me to look up what `fpm-cook inspect` actually does:
> prints a JSON representation of the recipe
so maybe `recipe_json` is better.

**Pull request merged**